DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs

```diff
@@ -302,6 +287,39 @@ public IActionResult EditDescription(EditDelegateGroupDescriptionViewModel model
             return RedirectToAction("Index");
         }
 
+        [HttpGet]
+        [Route("{groupId:int}/Courses/Add/{customisationId:int}")]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessGroup))]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessCourse))]
+        public IActionResult AddCourseToGroup(int groupId, int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!.Value;
+            var groupLabel = groupsService.GetGroupName(groupId, centreId)!;
+            var courseDetails = courseService.GetCourseDetailsForAdminCategoryId(customisationId, centreId, categoryId);
+            var supervisors = userService.GetSupervisorsAtCentreForCategory(centreId, courseDetails!.CourseCategoryId);
+            var viewModel = new AddCourseViewModel(groupId, customisationId, supervisors, groupLabel, courseDetails);
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("{groupId:int}/Courses/Add/{customisationId:int}")]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessGroup))]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessCourse))]
+        public IActionResult AddCourseToGroup(AddCourseFormData formData, int groupId, int customisationId)
+        {
+            if (!ModelState.IsValid)
+            {
+                var courseCategoryId = courseService.GetCourseCategoryId(customisationId);
+                var supervisors = userService.GetSupervisorsAtCentreForCategory(User.GetCentreId(), courseCategoryId);

```

Comment:
If `Supervisors` were included in the form data, would it get sent back automatically? From a code point of view, I think it's a bit better to send the list of supervisors back instead of fetching it anew. Means fewer lines of code in this method, and it's not particularly surprising or difficult to work out what's going on...

Reply:
Supervisors are the dropdown options, I'm pretty sure we don't want to be sending those back, otherwise what is the point of the FormData/ViewModel pattern.

Reply:
Ah, it doesn't get sent anyway. If they were being sent automatically it would have been more convenient to take them in - would have made this endpoint simpler.